### PR TITLE
Add IRIS Merchants connector

### PR DESCRIPTION
**Pull request by RolandHewage**
https://github.com/ballerina-platform/ballerina-extended-library/issues/64

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

JDK 11
Ballerina Beta 3
Ubuntu 20.04

https://www.iriscrm.com/
https://www.iriscrm.com/api/
https://www.iriscrm.com/api/#tag/Merchants
https://zapier.com/apps/iris-crm/integrations

### CODE REVIEW

File: openapi/iris.merchants/client.bal [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+    final readonly & map<string> apiKeys;
+    # Gets invoked to initialize the `connector`.
+    # The connector initialization requires setting the API credentials. 
+    # Create a [IRIS account](https://www.iriscrm.com) and obtain tokens by following [this guide](https://www.iriscrm.com/api/#section/Generate-an-API-token).
+    #
+    # + apiKeyConfig - Provide your API key as `X-API-KEY`. Eg: `{"X-API-KEY" : "<API key>"}` 
+    # + clientConfig - The configurations to be used when initializing the `connector` 
+    # + serviceUrl - URL of the target service 
+    # + return - An error if connector initialization failed 
+    public isolated function init(ApiKeysConfig apiKeyConfig, http:ClientConfiguration clientConfig =  {}, string serviceUrl = "/api/v1") returns error? {
```
Comment (aneeshafedo) [inline openapi/iris.merchants/client.bal:43]: Please provide a valid serviceURL.

Comment (RolandHewage) [reply openapi/iris.merchants/client.bal]: This is resolved now. Thank you.

File: openapi/iris.merchants/Module.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        X-API-KEY: "<API_KEY>"
+    }
+}
+merchants:Client baseClient = check new Client(config);
+```
+
+### Step 3: Invoke connector operation
+1. Now you can use the operations available within the connector. Note that they are in the form of remote operations.
+
+    Following is an example on how to get residuals templates using the connector.
```
Comment (sachinira) [inline openapi/iris.merchants/Module.md:39]: Check if the functionality mentioned is correct

Comment (RolandHewage) [reply openapi/iris.merchants/Module.md]: This is resolved now & the changes are amended. Thank you.

**Review by sachinira — APPROVED** [review]

**Review by SanduDS — APPROVED** [review]
Public record types' documentations can be improved more.

File: openapi/iris.merchants/client.bal [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/url;
+
+# Provides API key configurations needed when communicating with a remote HTTP endpoint.
+public type ApiKeysConfig record {|
+    # API keys related to connector authentication
+    map<string> apiKeys;
```
Comment (aneeshafedo) [inline openapi/iris.merchants/client.bal:23]: Implementation of `ApiKeysConfig` has been updated from the latest open-api tool. Please regenerate.

Comment (aneeshafedo) [inline openapi/iris.merchants/client.bal:23]: Implementation of `ApiKeysConfig` has been updated from the latest open-api tool. Please regenerate.

Comment (RolandHewage) [reply openapi/iris.merchants/client.bal]: This is regenerated now. Thank you.

Comment (RolandHewage) [reply openapi/iris.merchants/client.bal]: This is regenerated now. Thank you.

**Review by aneeshafedo — APPROVED** [review]

**Pull request merged**